commit 62b3b180ca125c37f1803dbde042829d83136e60
Author: Matthias Reichl <hias@horus.com>
Date:   Sun May 7 16:24:57 2017 +0200

    ASoC: bcm2835: Enforce full symmetry
    
    bcm2835's configuration registers can't be changed when a stream
    is running, which means asymmetric configurations aren't supported.
    
    Channel and rate symmetry are already enforced by constraints
    but samplebits had been missed.
    
    As hw_params doesn't check for symmetry constraints by itself
    and just returns success if a stream is running this led to
    situations where asymmetric configurations were seeming to
    succeed but of course didn't work because the hardware wasn't
    configured at all.
    
    Fix this by adding the missing samplerate symmetry constraint.
    
    Signed-off-by: Matthias Reichl <hias@horus.com>

diff --git a/sound/soc/bcm/bcm2835-i2s.c b/sound/soc/bcm/bcm2835-i2s.c
index 43f5715a0d5d..2e449d7173fc 100644
--- a/sound/soc/bcm/bcm2835-i2s.c
+++ b/sound/soc/bcm/bcm2835-i2s.c
@@ -783,7 +783,8 @@ static struct snd_soc_dai_driver bcm2835_i2s_dai = {
 				| SNDRV_PCM_FMTBIT_S32_LE
 		},
 	.ops = &bcm2835_i2s_dai_ops,
-	.symmetric_rates = 1
+	.symmetric_rates = 1,
+	.symmetric_samplebits = 1,
 };
 
 static bool bcm2835_i2s_volatile_reg(struct device *dev, unsigned int reg)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1129_62b3b180ca125c37f1803dbde042829d83136e60.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1129_62b3b180ca125c37f1803dbde042829d83136e60.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
